lunavl/sdk/estimators/face_estimators/ags.py

```diff
@@ -33,31 +31,78 @@ def __init__(self, coreEstimator: IAGSEstimatorPtr):
     def estimate(
-        self,
-        detection: Optional[FaceDetection] = None,
-        image: Optional[VLImage] = None,
-        boundingBox: Optional[BoundingBox] = None,
+        self, detection: Optional[FaceDetection] = None, imageWithFaceDetection: Optional[ImageWithFaceDetection] = None
     ) -> float:
         """
-        Estimate emotion on warp.
+        Estimate ags for single image/detection.
 
         Args:
-            image: image in R8G8B8 format
-            boundingBox: face bounding box of corresponding the image
             detection: face detection
+            imageWithFaceDetection: image with face detection
 
         Returns:
             estimated ags, float in range[0,1]
         Raises:
             LunaSDKException: if estimation failed
-            ValueError: if image and detection is Noee
+            ValueError: if image and detection are None
         """
         if detection is None:
-            if image is None or boundingBox is None:
-                raise ValueError("image and boundingBox or detection bust be not None")
-            error, ags = self._coreEstimator.estimate(image.coreImage, boundingBox.coreEstimation)
+            if imageWithFaceDetection is None:
+                raise ValueError("image and boundingBox or detection must be not None")
+            error, ags = self._coreEstimator.estimate(
+                imageWithFaceDetection.image.coreImage, imageWithFaceDetection.bBox.coreEstimation
+            )
         else:
             error, ags = self._coreEstimator.estimate(detection.image.coreImage, detection.boundingBox.coreEstimation)
+
         if error.isError:
             raise LunaSDKException(LunaVLError.fromSDKError(error))
         return ags
+
+    @CoreExceptionWrap(LunaVLError.EstimationAGSError)
+    def estimateAgsBatchByDetections(self, detections: List[FaceDetection]) -> List[float]:
+        """
+        Estimate ags for list of detections.
+
+        Args:
+            detections: face detection list
+
+        Returns:
+            list of estimated ags, float in range[0,1]
+        Raises:
+            LunaSDKException: if estimation failed
+            ValueError: if empty image list and empty detection list or images count not match bounding boxes count
+        """
+        coreImages = [detection.image.coreImage for detection in detections]
+        boundingBoxEstimations = [detection.boundingBox.coreEstimation for detection in detections]
+
+        validateInputByBatchEstimator(self._coreEstimator, coreImages, boundingBoxEstimations)
+        error, agsList = self._coreEstimator.estimate(coreImages, boundingBoxEstimations)
+
+        if error.isError:
+            raise LunaSDKException(LunaVLError.fromSDKError(error))
+        return agsList
+
+    @CoreExceptionWrap(LunaVLError.EstimationAGSError)
+    def estimateAgsBatchByImages(self, imageWithFaceDetectionList: List[ImageWithFaceDetection]) -> List[float]:
```

Comment:
Do you think that is better than 
```python
def estimateBatch(data: Union[List[ImageWithFaceDetection], List[FaceDetection]]):
```?
I think that we must add  function `estimateBatch` to  all estimators in the future.

Reply:
Union - may it be so

> add function estimateBatch to all estimators in the future

in the future - yes, for now - only if you insist

Reply:
I do. Because method `estimateBatch`  are added in this PR for several estimators. And estimateAgsBatchByImages is a strange exception.